web/server/codechecker_server/permissions.py

```diff
@@ -410,17 +417,16 @@ def _has_perm_impl(self, auth_names, are_groups=False):
             if not auth_names:
                 return False
 
+            # Compare authorization names in a case insensitive way.
+            auth_names_lower = [name.lower() for name in auth_names]
             ProdPerm = config_db_model.ProductPermission
             if are_groups:
-                # Compare group membership information in a case insensitive
-                # way.
-                auth_names = [name.lower() for name in auth_names]
                 query = self.__session. \
                     query(ProdPerm). \
                     filter(and_(
                         ProdPerm.permission == self._perm_name,
                         ProdPerm.product_id == self.__product_id,
-                        func.lower(ProdPerm.name).in_(auth_names),
+                        func.lower(ProdPerm.name).in_(auth_names_lower),
```

Comment:
The query in the if branch and else branch are the same. The only difference is in the `is_group.is_(True/False)` part. So no need for the if/else branch, just create one query and change the `ProdPerm.is_group.is_(True)` to `ProdPerm.is_group.is_(are_groups)`

Reply:
Done.